common/src/main/java/me/shedaniel/architectury/event/events/EntityEvent.java

```diff
@@ -65,26 +65,78 @@
     Event<PlaceBlock> PLACE_BLOCK = EventFactory.createInteractionResult();
     
     interface LivingDeath {
+        /**
+         * Invoked before a living entity dies.
+         * Equivalent to Forge's {@code LivingDeathEvent} event.
+         *
+         * @param entity The entity that is about to die.
+         * @param source Why the entity is about to die.
+         * @return Returning {@link InteractionResult#FAIL} prevents the entity from dying.
+         */
         InteractionResult die(LivingEntity entity, DamageSource source);
     }
     
     interface LivingAttack {
+        /**
+         * Invoked before an entity is hurt by a damage source.
+         * Equivalent to Forge's {@code LivingAttackEvent} event.
+         *
+         * <p>You currently cannot override the amount of damage the entity receives.
+         *
+         * @param entity The entity that is attacked.
+         * @param source The reason why the entity takes damage.
+         * @param amount The amount of damage the entity takes.
+         * @return Returning {@link InteractionResult#FAIL} prevents the entity from taking damage.
+         */
         InteractionResult attack(LivingEntity entity, DamageSource source, float amount);
     }
     
     interface LivingCheckSpawn {
+        /**
+         * Invoked before an entity is spawned into the world.
+         * This specifically concerns <i>spawning</i> through either a {@link BaseSpawner} or during world generation.
+         * Equivalent to Forge's {@code LivingSpawnEvent.CheckSpawn} event.
+         *
+         * @param entity  The entity that is about to spawn.
+         * @param world   The level the entity wants to spawn in.
+         * @param x       The x-coordinate of the spawn position.
+         * @param y       The y-coordinate of the spawn position.
+         * @param z       The z-coordinate the spawn position.
+         * @param type    The source of spawning.
+         * @param spawner The spawner. Can be {@code null}.
+         * @return Return {@link EventResult#interrupt(Boolean)} with true or false to let the entity spawn or prevent it.
```

Comment:
This doesn't have to be this specific.

Reply:
Maybe something in the middle like:

If the EventResult is *set*, the vanilla behaviour will be overridden.